server/matchmaker/map_pool.py

```diff
@@ -50,10 +50,22 @@ def choose_map(self, played_map_ids: Iterable[int] = ()) -> Map:
                 least_common = least_common[:i]
                 break
 
-        weights = [self.maps[id_].weight for id_, _ in least_common]
+        least_common_ids = {id_ for id_, _ in least_common}
 
-        map_id = random.choices(least_common, weights=weights, k=1)[0][0]
-        return self.maps[map_id].get_map()
+        # Anti-repetition is temporary disabled
+        # map_list = list((map.map_pool_map_version_id, map, 2 if (map.id in least_common_ids) and (vetoes_map.get(map.map_pool_map_version_id, 0) == 0) else 1) for map in self.maps.values())
+        map_list = list((map.map_pool_map_version_id, map, 1) for map in self.maps.values())
+        weights = [max(0, (1 - vetoes_map.get(id, 0) / max_tokens_per_map) * map.weight * least_common_multiplier) for id, map, least_common_multiplier in map_list]
+        return random.choices(map_list, weights=weights, k=1)[0][1]
 
     def __repr__(self) -> str:
         return f"MapPool({self.id}, {self.name}, {list(self.maps.values())})"
+
+
+class MatchmakerQueueMapPool(NamedTuple):
+    map_pool: MapPool
+    min_rating: int | None
```

Comment:
Maybe use `Optional[int]`, because looks like server hasn't adopted this syntax yet